crypto/impl_dispatch_test.cc

```diff
@@ -305,6 +308,55 @@ TEST_F(ImplDispatchTest, SHA3_512) {
         SHA3_512(in, 32, out);
       });
 }
+
+TEST_F(ImplDispatchTest, SHAKE256_Batched) {
+  // Assembly dispatch logic for Keccak-x4 on AArch64:
+  // - For Neoverse N1, we use scalar batched hybrid Keccak assembly from s2n-bignum
+  //   (`sha3_keccak4_f1600_alt()`) leveraging Neon and scalar assembly with
+  //   lazy rotations.
+  // - For Neoverse V1, V2, we use SIMD batched hybrid Keccak assembly from s2n-bignum
+  //   (`sha3_keccak4_f1600_alt2()`) leveraging Neon, Neon SHA3 extension,
+  //   and scalar assembly with lazy rotations.
+  // - Otherwise, if the Neon SHA3 extension is supported, we use the 2-fold
+  //   Keccak assembly from s2n-bignum (`sha3_keccak2_f1600()`) twice,
+  //   which is a straightforward implementation using the SHA3 extension.
+  // - Otherwise, fall back to four times the 1-fold Keccak implementation
+  //   with its own dispatch logic.
+  AssertFunctionsHit(
+      {
+          {kFlag_sha3_keccak4_f1600_alt,
+           have_s2n_bignum_asm_ && neoverse_n1_},
+          {kFlag_sha3_keccak4_f1600_alt2,
+           have_s2n_bignum_asm_ &&
+           (neoverse_v1_ || neoverse_v2_) &&
+           assembler_has_neon_sha3_extension_},
+          {kFlag_sha3_keccak2_f1600,
+           have_s2n_bignum_asm_ &&
+           !(neoverse_n1_ || neoverse_v1_ || neoverse_v2_) &&
+           (assembler_has_neon_sha3_extension_ && sha3_ext_)},
+	  // If we don't have assembly batched Keccak available,
```

Comment:
Indent is off for all the comment and the following code until #L352? Similar `TEST_F(ImplDispatchTest, SHA3_512)` #L291 to #L303. (Maybe the hidden tab issue).

Reply:
Fixed